test/podman-tests.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+# Do not set -e as we want to work on all results of `rpm -q dnf[5]`.
+set -exo pipefail
+
+if [ "$(id -u)" -ne 0 ];then
+    echo "Please run this script as superuser"
+    exit 1
+fi
+
+if [[ $1 == '' ]]; then
+    echo -e "Usage: podman-tests.sh TEST_TYPE STREAM\nTEST_TYPE can be 'e2e' or 'system'\nSTREAM can be 'upstream' or 'downstream'"
+    exit 1
+fi
+
+# `rhel` macro exists on RHEL, CentOS Stream, and Fedora ELN
+# `centos` macro exists only on CentOS Stream
+CENTOS_VERSION=$(rpm --eval '%{?centos}')
+RHEL_VERSION=$(rpm --eval '%{?rhel}')
+
+# For upstream tests, we need to test with podman and other packages from the
+# podman-next copr. For downstream tests (bodhi, errata), we don't need any
+# additional setup
+if [[ $2 == "upstream" ]]; then
+    # Use CentOS Stream 10 copr target for RHEL-10 until EPEL 10 becomes
+    # available
+    if [[ -n $CENTOS_VERSION || $RHEL_VERSION -ge 10 ]]; then
+        dnf -y copr enable rhcontainerbot/podman-next centos-stream-$CENTOS_VERSION
+    else
+        dnf -y copr enable rhcontainerbot/podman-next
+    fi
+    echo "priority=5" >> /etc/yum.repos.d/_copr:copr.fedorainfracloud.org:rhcontainerbot:podman-next.repo
+fi
+
+# Remove testing-farm repos if they exist
+rm -f /etc/yum.repos.d/tag-repository.repo
+
+# Install common dependencies for tests
+dnf -y install podman
+
+cat /etc/redhat-release
+rpm -q container-selinux podman
+
+if [[ $1 == "e2e" ]]; then
+    # Install dependencies for tests
+    dnf -y install golang
+    rpm -q golang
+
+    # /tmp is often unsufficient
+    export TMPDIR=/var/tmp
+
+    # dnf5 contains breaking changes
+    # Either of `dnf` OR `dnf5` will be installed, never both.
+    # To fetch srpm, dnf uses `--source`, dnf5 uses `--srpm`.
+    rpm -q dnf5 && SRPM_OPTS="--srpm" || SRPM_OPTS="--source"
+
+    # Fetch and extract latest podman source from the highest priority dnf repo
+    # NOTE: On upstream pull-requests, the srpm will be fetched from the
+    # podman-next copr while on bodhi updates, it will be fetched from Fedora's
+    # official repos.
+    dnf download $SRPM_OPTS podman
+    rpm2cpio podman*.src.rpm | cpio -di
+    tar zxf *.tar.gz
```

Comment:
EEEK. First: where are you cd'ed? This should be done from a scratch working dir, otherwise you don't know if you have space, or if you're vandalizing someone else's workspace, or if there are already existing `.tar.gz` files, or or or